File: SimpleBookAPI/Entities/FormControl.cs
using System.ComponentModel.DataAnnotations;

namespace SimpleBookAPI.Entities
{
  public class FormControl
  {
    public int Id { get; set; }
    public int FormId { get; set; }
    public Form Form { get; set; }
    public ICollection<FormControlOption> FormControlOptions { get; set; }
    public ICollection<FormControlValue> FormControlValues { get; set; }
    [Required]
    public string Type { get; set; }
    [Required]
    public string Name { get; set; }
    public string Description { get; set; }
    public string Placeholder { get; set; }
    [Required]
    public string Label { get; set; }
    public int Section { get; set; }
    public int Row { get; set; }
    public int Order { get; set; }
    public string CellSource { get; set; }
    [Required]
    public bool Required { get; set; }
  }
}

File: client/src/app/_models/formControl.ts
export interface FormControl {
  id: number;
  name: string;
  description: string;
  placeholder: string | null;
  label: string;
  section: number;
  row: number;
  order: number;
  cellSource: string;
  required: boolean;
  value: string | number | Date | boolean | undefined;
}

export interface FormSection {
  sectionNumber: number;
  rows: FormRow[];
}

export interface FormRow {
  rowNumber: number;
  controls: FormControl[];
}
